src/java/org/streamingpool/core/service/streamfactory/MergedErrorStreamFactory.java

```diff
@@ -0,0 +1,48 @@
+package org.streamingpool.core.service.streamfactory;
+
+import io.reactivex.Flowable;
+import org.reactivestreams.Publisher;
+import org.streamingpool.core.domain.DependencyGraph;
+import org.streamingpool.core.domain.ErrorStreamPair;
+import org.streamingpool.core.service.*;
+import org.streamingpool.core.service.diagnostic.ErrorStreamId;
+import org.streamingpool.core.service.impl.TrackKeepingDiscoveryService;
+import org.streamingpool.core.service.streamid.MergedErrorStreamId;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class MergedErrorStreamFactory implements StreamFactory {
+
+    private final InstrumentationService instrumentationService;
+
+    public MergedErrorStreamFactory(InstrumentationService instrumentationService) {
```

Comment:
Not possible to reused defined beans here ?

Reply:
what do you mean with this?

Reply:
Not sure it is relevant, but I saw on the [previous files](src/java/org/streamingpool/core/conf/EmbeddedPoolConfiguration.java) that you instantiate a `instrumentationService` bean. Could it be reused here with an `autowired` ? If not, could it be possible to define another bean and use it here ?

Reply:
apart from configuration/support classes we try not to bind classes to Spring with `@Autowired` (in other words, just bind as less classes as possible). Another advantage is that you could have a final field with constructor injection :)
Anyway, I think the correct way is to inject the `InstrumentationService` here. Did I get your point?